contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.2.0--2.3.0.sql

```diff
@@ -4176,6 +4176,139 @@ $$ LANGUAGE plpgsql IMMUTABLE STRICT;
 -- Drop the deprecated function
 CALL sys.babelfish_drop_deprecated_object('function', 'sys', 'get_tds_id_deprecated_2_3_0');
 
+ALTER VIEW sys.server_principals RENAME TO server_principals_deprecated_in_2_3_0;
```

Comment:
We should not rename the view in this case. We rename view/function/proc only if we cannot use `CREATE OR REPLACE`. For example, if output columns/types of the view are changed, then `CREATE OR REPLACE` would fail regardless of dependent objects. Otherwise, we should always try `CREATE OR REPLACE`.

Reply:
Also, please add an MVU test case where an object depends on `sys.server_principals`.

Reply:
Added.